Benchmark runs with config (runner)

SomeCuriosStudent opened the pull request:
## Does this close any issues?
Implements feature #22
## What does this PR do?
- added `run.py` to read a YAML config file via CLI `argparse` as command-line argument
    ```python
    parser = argparse.ArgumentParser(description="Run benchmarks using a YAML config file.")
    parser.add_argument("config", type=str, help="Path to the YAML configuration file.")
    args = parser.parse_args()
    ```

- added PyYaml libary as dependency in `requirement.txt` (`pyyaml>=6.0`)
- added `test_config.yaml` to test if `run.py` reads keys correctly
   ```python
   # Test Config for the run.py script
   task:
     name: test_task
     num_simulations: 100
     num_observations: 25
     num_posterior_samples: 50
   random_seed: 42
   method: NLE
   ```
- added code to extract and print all keys for testing purposes in the current `Process Task` code block:
   ```python
       # Process task
       task = config.get('task')
       if task is not None:
           task_name = task.get('name')
           # Get keys for testing purposes w/o defaults:
           num_simulations = task.get('num_simulations')
           num_observations = task.get('num_observations')
           num_posterior_samples = task.get('num_posterior_samples')
       else:
           task_name = "No task provided"

       method = config.get('method')

       print(
           f"Executing task: {task_name}\n"
           # printing keys for testing purposes
           f"num_simulations: {num_simulations}\n"
           f"num_observations: {num_observations}\n"
           f"num_posterior_samples: {num_posterior_samples}\n"
           f"method: {method}"
       )
   ```
## run.py
- checks for random_seed, if not 0 it will generate one
- extracts nested key task.name from config file correctly to print it
- logic for task execution isnt implemented, meaning the following keys are just dummies at the moment:
```yaml 
num_simulations: 100
num_observations: 25
num_posterior_samples: 50
method: NLE
``` 
- keys get extracted and printed by `run.py` for testting purposes
   ```
   > .../benchmark: python3 run.py test_config.yaml
   
   Using provided random seed: 42
   Executing task: test_task
   num_simulations: 100
   num_observations: 25
   num_posterior_samples: 50
   method: NLE
   ```
## Commands /  Syntax - Ubuntu example:
You can install PyYaml with `sudo apt-get install python3-yaml` for example
You can use runner with `python3 run.py test_config.yaml`, or the dictionary path for the yaml file

## ✅ Checklist

Put an `x` in the boxes that apply. If you're unsure about any of them, no worries—just ask!

- [x] I have read and followed the [contribution guidelines](https://github.com/sbi-dev/sbi/blob/main/CONTRIBUTING.md)
- [x] I have added helpful comments to my code where needed
- [x] I have added tests for new functionality
- [ ] (If applicable) I have reported how long new tests run and marked them with `pytest.mark.slow`

jmktpl reviewed (changes requested):
The basic structur of the runner is good and loading the yaml file works as well. I like that you print the keys to be able to check for mistakes in the .yaml or see that default cases were used.

 However, the logic for running a task and a simultor is still missing... I know that for now, we don't have any tasks ready yet, as it is another issue in this sprint, but it would be nice to have a rough structure of that logic already. That way we only need to adapt pramater passing or function names rather than implementing the whole logic. 
 
As a suggestion, it would make sense to introduce a simple task_registry containing all possible task classes implemented to this point (for now an empty registry), by importing all of those classes. If the name in the config file does not match the possible tasks, maybe you can raise an error saying that we need to check the .yaml file and giving all possible tasks as options. 

Running the simulator would be by calling <task_instance>().simulator(thetas), where thetas is a missspecified model. Maybe it would be better to implement it either almost like a dummy or just leave it out and tell @Eta-Epsilon52 to set up a new issue for the next sprint to fix the problems we have at this point. 

Except for this I noticed that negative inputs in the .yaml are also possible, want would be a problem when really running a task I think, so maybe try to add an option to use the default values when a negative entry was made.

SomeCuriosStudent commented:
> As a suggestion, it would make sense to introduce a simple task_registry containing all possible task classes implemented to this point (for now an empty registry), by importing all of those classes. If the name in the config file does not match the possible tasks, maybe you can raise an error saying that we need to check the .yaml file and giving all possible tasks as options.

Added `DummyTask` and `task_registry` with `DummyTask`, aswell as a hheck if the task exists in the `task_registry`.

> Running the simulator would be by calling <task_instance>().simulator(thetas), where thetas is a missspecified model. Maybe it would be better to implement it either almost like a dummy ...

Added task initiation and Dummy Simulator call (under "implement logic here" comment).

> Except for this I noticed that negative inputs in the .yaml are also possible, ...  maybe try to add an option to use the default values when a negative entry was made.

Added `validate_positive()` to check and correct to default value.

I tried to address your ideas as far as Im able to right now, see my newest commit: [2b5e0bf](https://github.com/sbi-misspecification-benchmark/sbi-misspecification-benchmark/pull/31/commits/2b5e0bf81d6adb5fa7601e1a850abf4717766187). Let me now with a review update if that's alright for now. I, for now, don't see how we could further add more detailed code without concrete tasks, datasets and the inference and simulator merged.
